### PR TITLE
Reworking team size check for switch

### PR DESCRIPTION
**Pull request by RedstoneFuture**
Fix and extend the team size check on the switch command.

### CODE REVIEW

File: missilewars-plugin/src/main/java/de/butzlabben/missilewars/game/Game.java [hunk 1]
```diff
@@ -800,16 +800,45 @@ public void updateGameInfo() {
...
+        
+        // Preventing an empty team when previously both teams had at least one player:
+        if ((currentTeamSize == 1) && (targetTeamSize >= 1)) return false;
+
+        int diff = getSmallerTeam().getEnemyTeam().getMembers().size() - getSmallerTeam().getMembers().size();
+
+        // max team difference: 30% (rounded) of target team size
+        float maxDiff = Math.round(targetTeamSize * 0.3);
+        // round up to at least 1:
+        if (maxDiff <= 0) maxDiff = 1;
```
Comment (daniel-naegele) [inline missilewars-plugin/src/main/java/de/butzlabben/missilewars/game/Game.java]: Use Math.min for this, maybe replace the line above with
`float maxDiff = Math.min(1, Math.round(targetTeamSize * 0.3));`

File: missilewars-plugin/src/main/java/de/butzlabben/missilewars/game/Game.java [hunk 2]
```diff
@@ -800,16 +800,45 @@ public void updateGameInfo() {
...
+
+        if (diff <= maxDiff) return true;
+        return false;
```
Comment (daniel-naegele) [inline missilewars-plugin/src/main/java/de/butzlabben/missilewars/game/Game.java]:
```suggestion
        return diff <= maxDiff;
```

File: missilewars-plugin/src/main/java/de/butzlabben/missilewars/game/Game.java (hunk 1 above)
Comment (RedstoneFuture) [reply missilewars-plugin/src/main/java/de/butzlabben/missilewars/game/Game.java]: Now, I used `Math.max()` to get the greater value of "1" and the math result. So, the min value is 1.

File: missilewars-plugin/src/main/java/de/butzlabben/missilewars/game/Game.java (hunk 2 above)
Comment (RedstoneFuture) [reply missilewars-plugin/src/main/java/de/butzlabben/missilewars/game/Game.java]: Oh, yea ... changed. :)

**Pull request merged**